device-registry/src/main/scala/org/genivi/sota/device_registry/DeviceRepository.scala

```diff
@@ -196,4 +225,28 @@ object Devices {
     _ <- systemInfos.insertOrUpdate(SystemInfo(id,data))
   } yield ()
 
+  def findGroupInfoByName(groupName: String, ns: Namespace)(implicit ec: ExecutionContext): DBIO[GroupInfoType] = {
+    groupInfos
+      .filter(r => r.namespace === ns && r.groupName === groupName)
+      .result
+      .headOption
+      .flatMap(_.fold[DBIO[GroupInfoType]](DBIO.failed(Errors.MissingGroupInfo))(x => DBIO.successful(x.groupInfo)))
+  }
+
+  def createGroupInfo(groupName: String, namespace: Namespace, data: GroupInfoType)(implicit ec: ExecutionContext)
+      : DBIO[Unit] =
+    for {
+      _ <- existsGroup(groupName, namespace).asTry.flatMap {
```

Comment:
instead of using a query to check for existance, we should relly on the database unique keys, rather thay rewriting them at the app level. This will then simplify our API, we just need to handle the SQL exceptiona nd transform it into `Errors.GroupInfoAlreadyExists`.

I already abstracted this handling on one of my branches, I will create a separate PR with that code so you can use it, but this means you'd change this to not use `existsGroup` and relly on db exceptions.